### PR TITLE
Add webimpress/http-middleware-compatibility support

### PR DESCRIPTION
**Pull request by Xerkus**
Add webimpress/http-middleware-compatibility support to make zend-mvc fully compatible with zend-stratigility 2.1

### CODE REVIEW

File: src/MiddlewareListener.php [hunk 1]
```diff
@@ -142,7 +143,10 @@ private function createPipeFromSpec(
             if (is_string($middlewareToBePiped) && $serviceLocator->has($middlewareToBePiped)) {
                 $middlewareToBePiped = $serviceLocator->get($middlewareToBePiped);
             }
-            if (! $middlewareToBePiped instanceof MiddlewareInterface && ! is_callable($middlewareToBePiped)) {
+            if (! $middlewareToBePiped instanceof MiddlewareInterface
```
Comment (michalbundyra) [inline src/MiddlewareListener.php:146]: We don't need this line anymore, please check line 11 in this file:
```php
use Interop\Http\ServerMiddleware\MiddlewareInterface;
```

it should be removed.

You are using MiddlewareInstance from `webimpress/http-middleware-compatibility` so it's enough.

Comment (Xerkus) [reply src/MiddlewareListener.php:146]: It is needed for stratigility 2.0.x

Comment (michalbundyra) [reply src/MiddlewareListener.php:146]: @Xerkus ok, fine, but we are not testing it then... Maybe we should set requirements to ^2.0.1 as it was and write additional test?

Comment (michalbundyra) [reply src/MiddlewareListener.php:146]: @Xerkus I've added tests in #262

Comment (Xerkus) [reply src/MiddlewareListener.php:146]: Not really _supporting_ it any more but allowing it to be used anyway as it is an optional dependency. If it was direct dependency, I would have bumped it to ^2.1.x and dropped that check.

Comment (michalbundyra) [reply src/MiddlewareListener.php:146]: As we keep this condition here, it should be also tested, this is why I've added it in PR #262 :)

**Comment by weierophinney** [comment]
I've written up a proposal for how we should go about updating to PSR-15 on the forums:

- https://discourse.zendframework.com/t/psr-15-compatibility-issues-and-proposal/378

Please read and comment there before moving forwards with the current pull request.

**Comment by Xerkus** [comment]
Since PSR-15 is out I am going to close this as won't fix.

**Pull request closed without merging**